File: src/Java/BehavioralPatterns/Mediator/ChainReaction/src/TextBox.java
public class TextBox extends Participant {
    private boolean isEnabled = false;

    public TextBox(Mediator mediator) {
        super(mediator);
    }

    public boolean isEnabled() {
        return isEnabled;
    }

    public void enable() {
        isEnabled = true;
        System.out.println("TextBox is enabled.");
        getMediator().onStateChanged(this);
    }

    public void disable() {
        isEnabled = false;
        System.out.println("TextBox is disabled.");
        getMediator().onStateChanged(this);
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/ChainReaction/textBox.js

const Participant = require("./participant");

class TextBox extends Participant {
    constructor(mediator) {
        super(mediator);
        this.isEnabled = false;
    }

    get isEnabled() {
        return this._isEnabled;
    }

    set isEnabled(value) {
        this._isEnabled = value;
    }

    enable() {
        this.isEnabled = true;
        console.log("TextBox is enabled.");
        this.mediator.onStateChanged(this);
    }

    disable() {
        this.isEnabled = false;
        console.log("TextBox is disabled.");
        this.mediator.onStateChanged(this);
    }
}

module.exports = TextBox;